website/content/docs/enterprise/pkcs11-provider/index.mdx

```diff
@@ -0,0 +1,307 @@
+---
+
+# PKCS#11 Provider
+
+[PKCS#11](http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html)
+is an open standard C API that provides a means to access cryptographic capabilities on a device.
+For example, it is often used to access a Hardware Security Module (HSM) (like a [Yubikey](https://www.yubico.com/)) from a local program (such as [GPG](https://gnupg.org/)).
+
+Vault provides a PKCS#11 library (or provider) so that Vault can be used as an SSM (Software Security Module).
+This allows a user to treat Vault like any other PKCS#11 device to manage keys, objects, and perform encryption and decryption in Vault using PKCS#11 calls.
+The PKCS#11 library connects to Vault's [KMIP Secrets Engine](/docs/secrets/kmip) to provide cryptographic operations and object storage.
+
+## Platform Support
+
+This library works with Vault Enterprise 1.11+ with the advanced data protection module in the license
+with the KMIP Secrets Engine.
+
+| Operating System | Architecture | Distribution      | glibc   |
+| ---------------- | -------------| ----------------- | ------- |
+| Linux            | x86-64       | RHEL 7 compatible | 2.17    |
+| Linux            | x86-64       | RHEL 8 compatible | 2.28    |
+| Linux            | x86-64       | RHEL 9 compatible | 2.34    |
+| macOS            | x86-64       | &mdash;           | &mdash; |
+
+The provider comes in the form of a shared C library, `libvault-pkcs11.so` (for Linux) or `libvault-pkcs11.dylib` (for macOS).
+It can be downloaded from [releases.hashicorp.com](https://releases.hashicorp.com/vault-pkcs11-provider).
+
+## Quick Start
+
+1. To use the provider, you will need access to a Vault Enterprise instance with the KMIP Secrets Engine.
+
+    For example, you can start one locally (if you have a license in the `VAULT_LICENSE` environment variable) with:
+
+    ```sh
+    docker pull hashicorp/vault-enterprise &&
+    docker run --name vault \
+      -p 5696:5696 \
+      -p 8200:8200 \
+      --cap-add=IPC_LOCK \
+      -e VAULT_LICENSE=$(printenv VAULT_LICENSE) \
+      -e VAULT_ADDR=http://127.0.0.1:8200 \
+      -e VAULT_TOKEN=root \
+      -e VAULT_ADDR=http://127.0.0.1:8200 \
+      hashicorp/vault-enterprise \
+      server -dev -dev-root-token-id root -dev-listen-address 0.0.0.0:8200
+    ```
+
+1. Configure the [KMIP Secrets Engine](/docs/secrets/kmip) and a KMIP *scope*. The scope is used to hold keys and objects.
+
+    ```sh
+    vault secrets enable kmip
+    vault write kmip/config listen_addrs=0.0.0.0:5696
+    vault write -f kmip/scope/my-service
+    vault write kmip/scope/my-service/role/admin operation_all=true
+    vault write -f -format=json kmip/scope/my-service/role/admin/credential/generate | tee kmip.json
+    ```
+
+    ~> **Important**: When configuring KMIP in production, you will probably need to set the
+    `server_hostnames` and `server_ips` [configuration parameters](https://www.vaultproject.io/api-docs/secret/kmip#parameters),
```

Comment:
I think this can use a relative link:

```suggestion
    `server_hostnames` and `server_ips` [configuration parameters](/api-docs/secret/kmip#parameters),
```